common/wuanalysis/anarule.cpp

```diff
@@ -165,11 +165,39 @@ class IoSkewRule : public AActivityRule
     const char * category;
 };
 
+class KeyedJoinExcessRejectedRowsRule : public ActivityKindRule
+{
+public:
+    KeyedJoinExcessRejectedRowsRule() : ActivityKindRule(TAKkeyedjoin) {}
+
+    virtual bool check(PerformanceIssue & result, IWuActivity & activity, const WuAnalyseOptions & options) override
+    {
+        stat_type preFiltered = activity.getStatRaw(StNumPreFiltered);
+        if (preFiltered)
+        {
+            IWuEdge * inputEdge = activity.queryInput(0);
+            stat_type rowscnt = inputEdge->getStatRaw(StNumRowsProcessed);
+            stat_type preFilteredPer = statPercent( preFiltered*100 ) / rowscnt;
```

Comment:
@ghalliday Any possibility for inputEdge->getStatRaw(StNumRowsProcessed to return 0 here?

Reply:
@shamser I would say very unlikely, but probably best to protect against division by zero just in case.  (E.g. a bug in the engine meaning that a row count wasn't added to an edge.)